instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RefreshTtlTest.kt

```diff
@@ -0,0 +1,600 @@
+        createMapboxNavigation(frequentRefreshOptions)
+        mockWebServerRule.requestHandlers.clear()
+        mockWebServerRule.requestHandlers.add(
+            MockDirectionsRequestHandler(
+                "driving-traffic",
+                readRawFileText(context, R.raw.route_response_route_refresh_with_ttl),
+                coordinates
+            )
+        )
+        val routeOptions = generateRouteOptions(coordinates)
+        mapboxNavigation.startTripSession()
+        stayOnPosition(coordinates[0].latitude(), coordinates[0].longitude(), 190f) {
+            val requestedRoutes = mapboxNavigation.requestRoutes(routeOptions)
+                .getSuccessfulResultOrThrowException()
+                .routes
+
+            mapboxNavigation.setNavigationRoutes(requestedRoutes)
+            val routesInvalidatedResult = withTimeout(4000) {
+                mapboxNavigation.routesInvalidatedResults().first()
+            }
+            assertEquals(requestedRoutes, routesInvalidatedResult)
+        }
+    }
+
+    @Test
+    fun refreshTtlExpiresOnDifferentRefreshesForDifferentRoutes() = sdkTest {
+        createMapboxNavigation(frequentRefreshOptions)
+        mockWebServerRule.requestHandlers.clear()
+        mockWebServerRule.requestHandlers.add(
+            MockDirectionsRequestHandler(
+                "driving-traffic",
+                readRawFileText(context, R.raw.route_response_route_refresh_with_different_ttls),
+                coordinates
+            )
+        )
+        mockWebServerRule.requestHandlers.add(
+            MockDirectionsRefreshHandler(
+                "route_response_route_refresh_with_different_ttls",
+                readRawFileText(context, R.raw.route_response_route_refreshed_with_different_ttls),
+            )
+        )
+        val routeOptions = generateRouteOptions(coordinates)
+        mapboxNavigation.startTripSession()
+        stayOnPosition(coordinates[0].latitude(), coordinates[0].longitude(), 190f) {
+            val requestedRoutes = mapboxNavigation.requestRoutes(routeOptions)
+                .getSuccessfulResultOrThrowException()
+                .routes
+
+            mapboxNavigation.setNavigationRoutes(requestedRoutes)
+
+            val invalidatedResults = mutableListOf<List<NavigationRoute>>()
+            mapboxNavigation.registerRoutesInvalidatedObserver {
+                invalidatedResults.add(it)
+            }
+            mapboxNavigation.routesUpdates().filter {
+                it.reason == RoutesExtra.ROUTES_UPDATE_REASON_REFRESH
+            }.first()
+            assertEquals(listOf(requestedRoutes.first()), invalidatedResults[0])
+
+            withTimeout(4000) {
+                mapboxNavigation.routesInvalidatedResults().first()
+            }
+            assertEquals(listOf(requestedRoutes[1].id), invalidatedResults[1].map { it.id })
+        }
+    }
+
+    @Test
+    fun continuousAlternativeRouteIsInvalidated() = sdkTest {
```

Comment:
the route refresh and continuous alternatives features seems to be independent from each other. Why did you decide to test this case? Probably there is some dependency, isn't it?

Reply:
I found the answer in the implementation 🙂